oms/oms.go

```diff
@@ -56,57 +52,13 @@ func NewOmsLogger(customerId, sharedKey, logTypePrefix LogType) *OmsLogger {
 }
 
 func (o *OmsLogger) Write(p []byte) (n int, err error) {
-	matches := o.GinLogMatcher.FindStringSubmatch(string(p))
-	if len(matches) > 0 {
-		httpStatus, latency := parseFromGinLog(matches)
-		return o.writeLogs(p, o.LogTypes[1], WithHttpStatus(httpStatus), WithLatency(latency))
+	ginLog, err := NewGinLog(p)
+	if err == nil {
+		return o.writeLogs(p, o.LogTypes[1], WithHttpStatus(ginLog.HttpStatus), WithLatency(ginLog.Latency))
```

Comment:
how about using:
`return o.writeLogs(p, o.LogTypes[1], ginLog.Data)`
meaning the `GinLog` would already marshal the JSON.
Then the logEntry doesn't need any specific information like latency or http path